dotcom-rendering/src/components/WeatherWrapper.importable.tsx

```diff
@@ -1,36 +1,19 @@
 import type { EditionId } from '../lib/edition';
-import type { Tuple } from '../lib/tuple';
 import { useApi } from '../lib/useApi';
+import type { WeatherApiData } from '../types/weather';
 import { Weather, WeatherPlaceholder } from './Weather';
 
-/**
- * Our weather API returns 24 forecast.
- * Each forecast is 1 hour offset from the previous forecast, and the first forecast is 1 hour offset from Now.
- */
-export type WeatherForecast = [
-	...Tuple<WeatherData, 12>,
-	...Tuple<WeatherData, 12>,
-];
+const appendPartnerCodeToUrl = (
+	url: string | undefined,
+): string | undefined => {
+	if (!url) {
+		return undefined;
+	}
 
-export type WeatherData = {
-	description: string;
-	icon: number;
-	link?: string;
-	dateTime?: string;
-	temperature: {
-		metric: number;
-		imperial: number;
-	};
-};
+	const link = new URL(url);
+	link.searchParams.append('partner', 'web_guardian_adc');
 
-export type WeatherApiData = {
-	location: {
-		id: string;
-		city: string;
-		country: string;
-	};
-	weather: WeatherData;
-	forecast: WeatherForecast;
+	return link.toString();
```

Comment:
I think [elsewhere in this repo](https://github.com/search?q=repo%3Aguardian%2Fdotcom-rendering+.href&type=code) we tend to use [URL.href](https://developer.mozilla.org/en-US/docs/Web/API/URL/href). They are identical, though.

Reply:
Thanks. I like `URL.href`